### PR TITLE
Plated Catwalk Fixes, And Construction

### LINKED ISSUE #1527: Removing plating with an open catwalk above it removes the catwalk
## Round ID:

2725

## Reproduction:

1. Place catwalk above flooring
2. Open the catwalk to access the floor
3. Crowbar off the plating or unbolt the reinforced floor (only tested with reinf floor so far)
4. Your catwalk is now gone

### PR DESCRIPTION
**Pull request by Ikalpo**
## About The Pull Request
fixes #1527
`PREVENT_CLICK_UNDER_1` now doesn't need density to work, so it's good to use with catwalks.
using `PREVENT_CLICK_UNDER_1`, you can now no longer remove pipes or cables under catwalks
fixes regular "over" catwalks destroying cables when deconstructed
fixes techfloor grid tiles not showing up in the ctrl-click menu for plasteel tiles
adjusted the color of dark plated catwalks
adjusted the plated catwalk sprites to smooth better
fixed plated catwalks icon after opening and closing

**This should be added to the wiki**
plated catwalks can be created by using rods on monotiles, and deconstruct back into monotiles and rods with wirecutters

## Why It's Good For The Game

FIX MAN GOOD
also it sucks that these were irreplaceable

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

image
image

</details>

## Changelog
:cl:
add: Plated catwalks can now be created with rods and monotiles
fix: Fixed being able to interact with objects under catwalks
fix: Fixed plated catwalks being destroyed when pulling up a floor tile under them
imageadd: Plated catwalks have new sprites to smooth better
imagedel: Got rid of the dumb non-smoothed sprite when a plated catwalk is closed
fix: Fixes techfloor grid tiles not showing up in the ctrl-click menu for plasteel tiles
/:cl:

### CODE REVIEW

File: nsv13/code/game/objects/structures/catwalk.dm [hunk 1]
```diff
@@ -38,13 +40,25 @@
...
+/obj/structure/lattice/catwalk/over/ship/deconstruct()
+	new plating_type(get_turf(src))
+	. = ..()
+
+/obj/structure/lattice/catwalk/over/attackby(obj/item/C, mob/user, params)
+	if(resistance_flags & INDESTRUCTIBLE)
+		return
+	if(C.tool_behaviour == TOOL_WIRECUTTER)
+		to_chat(user, "<span class='notice'>Slicing [name] joints ...</span>")
+		deconstruct()
```
Comment (Bokkiewokkie) [inline nsv13/code/game/objects/structures/catwalk.dm:64]:
```suggestion
/obj/structure/lattice/catwalk/over/attackby(obj/item/C, mob/user, params)
	if(resistance_flags & INDESTRUCTIBLE)
		return ..()
	if(C.tool_behaviour == TOOL_WIRECUTTER)
		to_chat(user, "<span class='notice'>Slicing [name] joints ...</span>")
		deconstruct()
	else
		return ..()

```
Still needs to be able to get attacked.
Cable laying is done with can_lay_cable() so  (in [code/game/turfs/turf.dm](https://github.com/BeeStation/NSV13/blob/5232787ff4630796e07e1099373ea4af669ae032/code/game/turfs/turf.dm#L335))

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]

File: nsv13/code/game/objects/structures/catwalk.dm (hunk 1 above)
Comment (Ikalpo) [reply nsv13/code/game/objects/structures/catwalk.dm:64]: Regular catwalks, including the `/over/` type, override attackby() just like that and it's fine.
No idea what you are talking about with can_lay_cable(), cable laying works just fine in every situation I tested.

Comment (Bokkiewokkie) [reply nsv13/code/game/objects/structures/catwalk.dm:64]: Alright then ig, if it works it works

**Review by Bokkiewokkie — APPROVED** [review]

**Review by covertcorvid — APPROVED** [review]

**Review by Bokkiewokkie — APPROVED** [review]
Really hope there's not another structure we missed

**Pull request merged**